Enum for abstainVote

lhsunyc opened the pull request:
# Description
This PR fixes #230 

Create the abstain vote enum and extract it to the common folder and apply changes to relevant files.
# Demo
please refer to files changed


# Type of change:
* Bug fix (Non-breaking change which fixes a bug)

lhsunyc committed:
Add an enum for abstain vote

"abstain vote" is renamed to "no confidence vote" to prevent ambiguous meaning.

lhsunyc committed:
Extract the abstain vote enum to common folder and apply changes to relevant files

lhsunyc committed:
Reset changes applied in front-end files

JChamberlain28 commented on backend/domain/DTO/response/ListVoteDTO.js:
```diff
@@ -1,8 +1,11 @@
+const evotes = require('../../../../common/Enum/Vote');
```
why is the name here 'evotes' but in other files, it is 'votes'. Also, a more descriptive name for this variable might be useful. What is it representing? 'votes' is semi ambiguous.

milkyway404 replied:
I agree. Perhaps VoteType?

JChamberlain28 commented on backend/domain/VoteHandler.js:
```diff
@@ -3,6 +3,8 @@
  * This includes tallying votes, and figuring out who the most voted player is.
  * The vote handler keeps track of vote maps. These are JSON objects where each key is a player nickname, and each value is the Player object that was voted for.
  */
+const votes = require('../../common/Enum/Vote');
```
more desriptive name.

JChamberlain28 reviewed (changes requested):
Consider these changes. Otherwise, it resolves the issue. Tested working on my computer.

lhsunyc committed:
Rename variable "votes" to "voteType" to avoid ambiguous meaning

yzho976 reviewed (dismissed):
These changes looks good to me. One thing Im not sure is why the code for TrialStateChangeEvents.js has been deleted and added again. But that's acceptable.

JChamberlain28 reviewed (dismissed):
Thanks for adding the changes.
LGTM

Before merging you will need to ensure all tests pass though. There are 2 back end test failures.

lhsunyc committed:
Merge branch 'a2-main' into main

lhsunyc committed:
Add a space in VoteEnum value

Also, fix eslint issue in NightTimeVoteEvents.js

ambiyu commented on backend/Events/GameStateEvents/TrialStateChangeEvents.js:
```diff
@@ -1,74 +1,75 @@
+    const playerChosen = room.voteHandler.getTrialVotedPlayer();
+
+    if (playerChosen && playerChosen !== voteType.NoConfidenceVote) {
+        room.getPlayerByNickname(playerChosen).isAlive = PlayerStatus.KILLED_BY_TOWN;
```
Not sure why this was changed to `isAlive`? Please change it back to `status`:
```
       room.getPlayerByNickname(playerChosen).status = PlayerStatus.KILLED_BY_TOWN;
```

lhsunyc replied:
The problem solved, thanks andy!

ambiyu reviewed (changes requested):
Thanks for this. Resolving the requested change should fix the tests failing. Please also change all instances of voteType to VoteType (uppercase V) to keep enum naming consistent.

lhsunyc committed:
Fix the issue in TrialStateChangeEvents.js

Also, changed all instances of voteType to VoteType to keep enum naming consistent.

yzho976 reviewed (approved):
This looks good for me now.

ambiyu reviewed (approved):
LGTM now. Thanks for the changes

JChamberlain28 committed:
Merge branch 'a2-main' into main

khal745 reviewed (changes requested):
Looking good on the backend, but I think some frontend changes are required given we are using this new enum. Have a look at #291 where the `Table.jsx` and `useGameState.js` file required some small updates so they pass around the newly changed info. Because its the same string ('no Confidence') used in frontend compared to the enum value, I think the game should work, but passing the literal string in frontend doesn't solve the original issue. We could merge this if we need to though.

khal745 reviewed (approved):
For now we can merge and make a follow up PR to address the frontend issues.
